app/shared/app-data/src/commonMain/kotlin/domain/settings/ProxyProvider.kt

```diff
@@ -83,9 +84,16 @@ class SystemProxyProvider(
 class SettingsBasedProxyProvider(
     private val settingsRepository: SettingsRepository,
     backgroundScope: CoroutineScope,
+) : ProxyProvider by ProxySettingsFlowProxyProvider(
+    settingsRepository.proxySettings.flow, backgroundScope,
+)
```

Comment:
use a constructor-like a function to construct a ProxySettingsFlowProxyProvider instead of making a type.